src/Resolvers/Microsoft.DotNet.NativeWrapper/NETCoreSdkResolverNativeWrapper.cs

```diff
@@ -50,5 +50,32 @@ public static string[] GetAvailableSdks(string dotnetExeDirectory)
 
             return list.Entries;
         }
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct hostfxr_initialize_parameters
+        {
+
+        }
+
+        public static int InitializeForRuntimeConfig(string runtimeConfigPath)
+        {
+            var result = -1;
+            IntPtr hostContextHandle = default;
+
+            hostfxr_initialize_parameters parameters = new hostfxr_initialize_parameters();
+
+            IntPtr parametersPtr = Marshal.AllocHGlobal(Marshal.SizeOf(parameters));
+            Marshal.StructureToPtr(parameters, parametersPtr, false);
+
+            if (File.Exists(runtimeConfigPath))
+            {
+                result = Interop.RunningOnWindows
+                    ? Interop.Windows.hostfxr_initialize_for_runtime_config(runtimeConfigPath, default, out hostContextHandle)
+                    : Interop.Unix.hostfxr_initialize_for_runtime_config(runtimeConfigPath, default, out hostContextHandle);
+            }
+
+            Marshal.FreeHGlobal(parametersPtr);
+            return result;
+        }
```

Comment:
This is a coding nit - if we know the different result codes, can those be turned into a enum and returned from this method instead of an int?